packages/smart-wallet/src/smartWallet.js

```diff
@@ -156,405 +145,463 @@ export const initState = (unique, shared) => {
-    // Payments that couldn't be deposited when received.
-    // NB: vulnerable to uncapped growth by unpermissioned deposits.
-    paymentQueues: makeScalarBigMapStore('payments queues', {
-      durable: true,
-    }),
-    // Invitation amounts to save for persistent lookup
-    offerToUsedInvitation: makeScalarBigMapStore(
-      'invitation amounts by offer',
-      {
-        durable: true,
-      },
-    ),
-    // Invitation makers yielded by offer results
-    offerToInvitationMakers: makeScalarBigMapStore(
-      'invitation makers by offer',
-      {
-        durable: true,
-      },
-    ),
-    // Public subscribers yielded by offer results
-    offerToPublicSubscriberPaths: makeScalarBigMapStore(
-      'public subscribers by offer',
-      {
+  /**
+   *
+   * @param {UniqueParams} unique
+   * @returns {State}
+   */
+  const initState = unique => {
+    // Some validation of inputs.
+    mustMatch(
+      unique,
+      harden({
+        address: M.string(),
+        bank: M.eref(M.remotable()),
+        invitationPurse: PurseShape,
+        currentStorageNode: M.eref(StorageNodeShape),
+        walletStorageNode: M.eref(StorageNodeShape),
+      }),
+    );
+
+    const preciousState = {
+      // Private purses. This assumes one purse per brand, which will be valid in MN-1 but not always.
+      brandPurses: makeScalarBigMapStore('brand purses', { durable: true }),
+      // Payments that couldn't be deposited when received.
+      // NB: vulnerable to uncapped growth by unpermissioned deposits.
+      paymentQueues: makeScalarBigMapStore('payments queues', {
         durable: true,
-      },
-    ),
-  };
-
-  const nonpreciousState = {
-    // What purses have reported on construction and by getCurrentAmountNotifier updates.
-    purseBalances: makeScalarMapStore(),
-    /** @type {StoredPublishKit<UpdateRecord>} */
-    updatePublishKit: harden(
-      makeStoredPublishKit(myWalletStorageNode, shared.publicMarshaller),
-    ),
-    /** @type {StoredPublishKit<CurrentWalletRecord>} */
-    currentPublishKit: harden(
-      makeStoredPublishKit(myCurrentStateStorageNode, shared.publicMarshaller),
-    ),
+      }),
+      // Invitation amounts to save for persistent lookup
+      offerToUsedInvitation: makeScalarBigMapStore(
+        'invitation amounts by offer',
+        {
+          durable: true,
+        },
+      ),
+      // Invitation makers yielded by offer results
+      offerToInvitationMakers: makeScalarBigMapStore(
+        'invitation makers by offer',
+        {
+          durable: true,
+        },
+      ),
+      // Public subscribers yielded by offer results
+      offerToPublicSubscriberPaths: makeScalarBigMapStore(
+        'public subscribers by offer',
+        {
+          durable: true,
+        },
+      ),
+    };
+
+    /** @type {PublishKit<UpdateRecord>} */
+    const updatePublishKit = makeWalletPublishKit();
+    // NB: state size must not grow monotonically
```

Comment:
I'm not sure what "must not grow monotically" means.  (I realize you moved it.)   Do you think it's trying to say that it "must grow sub-linearly"?

Reply:
I wrote the moved string. I mean it must not continue to grow; it must also drop information that's no longer useful.

If you propose a better way to say that I can drop it into another PR

Reply:
"state size must not grow without bound" or "state must be limited in size".  Not sure which you mean.

"must not grow monotonically" sounds to me like it can reach whatever size it wants, as long as it occasionally pares things back somewhat.